test/abs-reshape.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 Google LLC
```

Comment:
Do we want to add more structure around dynamic shape tests? Like Upperbound, retriggering-mem-planning, illegal inputs etc. A harness of sorts may be? I haven't thought enough about this TBH.

Reply:
More testing will be needed. Upper bounding is already tested for by verifying that xnn_status_reallocation_required is returned when the new size exceeds the maximum size given